test/functional/feature_bind_extra.py

```diff
@@ -37,37 +37,41 @@ def setup_network(self):
         loopback_ipv4 = addr_to_hex("127.0.0.1")
 
         # Start custom ports by reusing unused p2p ports
-        port = p2p_port(self.num_nodes)
+        def extra_port():
+            port = p2p_port(extra_port.index)
+            extra_port.index += 1
+            return port
+        extra_port.index = self.num_nodes
```

Comment:
If you apply the patch below, the patch above will work.
Is there a reason why the starting port index should be the same as the number of nodes?

```suggestion
        extra_port.index = 0
```